pkg/apis/core/v1beta1/types_cloudprofile.go

```diff
@@ -164,12 +166,24 @@ type ExpirableVersion struct {
 	// Version is the version identifier.
 	Version string `json:"version" protobuf:"bytes,1,opt,name=version"`
 	// ExpirationDate defines the time at which this version expires.
+	// Deprecated: Is replaced by Lifecycle.
 	// +optional
 	ExpirationDate *metav1.Time `json:"expirationDate,omitempty" protobuf:"bytes,2,opt,name=expirationDate"`
 	// Classification defines the state of a version (preview, supported, deprecated).
-	// To get the currently valid classification, use CurrentLifecycleClassification().
+	// Deprecated: Is replaced by Lifecycle.
 	// +optional
 	Classification *VersionClassification `json:"classification,omitempty" protobuf:"bytes,3,opt,name=classification,casttype=VersionClassification"`
+	// Lifecycle defines the lifecycle stages for this version. Cannot be used in combination with Classification and ExpirationDate.
+	// +optional
+	Lifecycle []LifecycleStage `json:"lifecycle,omitempty" protobuf:"bytes,4,opt,name=lifecycle"`
+}
+
+type LifecycleStage struct {
```

Comment:
Requires docstring. Example:
```suggestion
// LifecycleStage describes a stage in the classification lifecycle.
type LifecycleStage struct {
```